modules/porous_flow/src/auxkernels/PorousFlowDarcyVelocityComponentLowerDimensional.C

```diff
@@ -51,14 +46,22 @@ PorousFlowDarcyVelocityComponentLowerDimensional::computeValue()
                "only since it employs "
                "PorousFlowDarcyVelocityComponentLowerDimensional\n");
 
+  // MOOSE will automatically make grad(P) lie along the element's tangent direction
+  // but we need to project gravity in that direction too
+
+  // tang_xi is the element's tangent vector in xi direction
+  const std::vector<RealGradient> tang_xi = _assembly.getFE(FEType(), elem_dim)->get_dxyzdxi();
   RealVectorValue tangential_gravity =
-      (_gravity * (*_tang_xi[elem_dim])[_qp] / (*_tang_xi[elem_dim])[_qp].norm_sq()) *
-      (*_tang_xi[elem_dim])[_qp];
+      (_gravity * tang_xi[_qp] / tang_xi[_qp].norm_sq()) * tang_xi[_qp];
   if (elem_dim == 2)
-    tangential_gravity +=
-        (_gravity * (*_tang_eta[elem_dim])[_qp] / (*_tang_eta[elem_dim])[_qp].norm_sq()) *
-        (*_tang_eta[elem_dim])[_qp];
+  {
+    // tang_eta is the element's tangent vector in eta direction
+    const std::vector<RealGradient> tang_eta = _assembly.getFE(FEType(), elem_dim)->get_dxyzdeta();
```

Comment:
Reference here too

Reply:
Holy Dooley you've got a good eye, @permcody .  Thanks.